be/src/storage/rowset/segment.cpp

```diff
@@ -225,8 +238,13 @@ StatusOr<ChunkIteratorPtr> Segment::_new_iterator(const Schema& schema, const Se
             continue;
         }
         if (!_column_readers[column_id]->segment_zone_map_filter(pair.second)) {
-            read_options.stats->segment_stats_filtered += _column_readers[column_id]->num_rows();
-            return Status::EndOfFile(strings::Substitute("End of file $0, empty iterator", _fname));
+            // skip segment zonemap filter when this segment has column files link to it.
+            if (_use_segment_zone_map_filter(read_options)) {
```

Comment:
I do this on purpose because  `_use_segment_zone_map_filter` is heavier than `_column_readers[column_id]->segment_zone_map_filter`. `_use_segment_zone_map_filter` could read from disk when BE is just start.